### ISSUE TITLE
dotnet msbuild hanging

### ISSUE THREAD

**Issue by seancpeters** [Microsoft/msbuild#3227]
_From @fuocor on April 24, 2018 18:12_

building using the cli is hanging at completion on Windows Server.

This is the abbreviated output (full output is attached).
[build.zip](https://github.com/dotnet/templating/files/1944017/build.zip)

```
 1:2>Done building target "IncrementalClean" in project "Ossiaco.Common.csproj".
     Target "PostBuildEvent" skipped, due to false condition; ('$(PostBuildEvent)' != '' and ('$(RunPostBuildEvent)' != 'OnOutputUpdated' or '$(_AssemblyTimestampBeforeCompile)' != '$(_AssemblyTimestampAfterCompile)')) was evaluated as ('' != '' and ('' != 'OnOutputUpdated' or '' != '')).
     Target "GenerateBuildDependencyFile" skipped. Previously built successfully.
     Target "GenerateBuildRuntimeConfigurationFiles" skipped, due to false condition; ( '$(GenerateRuntimeConfigurationFiles)' == 'true') was evaluated as ( '' == 'true').
 1:2>Target "CoreBuild" in file "C:\Program Files\dotnet\sdk\2.1.300-preview3-008427\Microsoft.Common.CurrentVersion.targets" from project "C:\ossiaco\dotnet\src\Ossiaco.Common\Ossiaco.Common.csproj" (target "Build" depends on it):
 1:2>Done building target "CoreBuild" in project "Ossiaco.Common.csproj".
 1:2>Target "AfterBuild" in file "C:\Program Files\dotnet\sdk\2.1.300-preview3-008427\Microsoft.Common.CurrentVersion.targets" from project "C:\ossiaco\dotnet\src\Ossiaco.Common\Ossiaco.Common.csproj" (target "Build" depends on it):
 1:2>Done building target "AfterBuild" in project "Ossiaco.Common.csproj".
     Target "_CheckForInvalidConfigurationAndPlatform" skipped. Previously built successfully.
     Target "_CheckForInvalidConfigurationAndPlatform" skipped. Previously built successfully.
 1:2>Target "Build" in file "C:\Program Files\dotnet\sdk\2.1.300-preview3-008427\Microsoft.Common.CurrentVersion.targets" from project "C:\ossiaco\dotnet\src\Ossiaco.Common\Ossiaco.Common.csproj" (entry point):
 1:2>Done building target "Build" in project "Ossiaco.Common.csproj".
     Target "_PackAsBuildAfterTarget" skipped, due to false condition; ('$(GeneratePackageOnBuild)' == 'true' AND '$(IsInnerBuild)' != 'true') was evaluated as ('false' == 'true' AND 'true' != 'true').
 1:2>Done Building Project "C:\ossiaco\dotnet\src\Ossiaco.Common\Ossiaco.Common.csproj" (Build target(s)).
```

It requires a Ctrl+C to break out of the build. Here is the dotnet --info

```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300-preview3-008427
 Commit:    423010bf0d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300-preview3-008427\

Host (useful for support):
  Version: 2.1.0-preview3-26411-06
  Commit:  8faa8fcfcf

.NET Core SDKs installed:
  2.1.300-preview3-008427 [C:\Program Files\dotnet\sdk]
  2.2.100-preview1-008633 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0-preview2-30431 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.0-preview2-30475 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0-preview2-30431 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.0-preview2-30475 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0-preview2-26314-02 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0-preview3-26411-06 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

 

_Copied from original issue: dotnet/templating#1519_

**Comment by rainersigwald** [comment]
This log is odd in a few ways.

* SDK `2.1.300-preview3-008427` is not a version I'd expect; maybe it's an old CLI master-branch build?
* MSBuild `15.7.124.19828` is fairly old now, predating some node-reuse changes.
* The log indicates that the project is multitargeted to `netstandard2.0;netcoreapp2.1`, but only the `netstandard2.0` version completes. That explains the hang--MSBuild is waiting on the `netcoreapp2.1` build.

I suspect a failure to launch or connect to a worker node

@fuocor can you try a few things?

* Does this repro with [the preview2 SDK](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300-preview2)?
* If so, can you use Process Monitor to take a trace of process start and process exit events during the build?
* Can you also set two debugging flags for MSBuild and report the traces?

```
set MSBUILDDEBUGCOMM=1
mkdir c:\msbuildcommlogs
set MSBUILDDEBUGPATH=c:\msbuildcommlogs

dotnet build
```

You can use a different folder, of course--and please zip up everything from that folder and attach it.

**Comment by fuocor** [comment]
Thanks for the quick response...

- Yes it builds with the preview 2 sdk

here is the zip containing the process & build logs.

I did notice while monitoring the processes, that the build tries restarting msbuild on a regular interval


[MSBuild_CommTrace_PID_7608.zip](https://github.com/Microsoft/msbuild/files/1947107/MSBuild_CommTrace_PID_7608.zip)

**Comment by fuocor** [comment]
the problem appears to be related to node... I am building on a windows machine using nvs to target chackracore and node. 

Rerunning nvs and selecting another version appears to have resolved the issue and reselecting chackracore works as well.

rich

**Comment by rainersigwald** [comment]
Thanks for checking it out. This is an instance of #3142, which was fixed for preview 2 and up. The symptoms match. It is indeed trying to launch MSBuild every 30 seconds--it'll give up eventually after trying 10 times.

Please let us know if you see this on a preview2+ build and we can reopen and investigate further.

**Comment by rainersigwald** [comment]
Duplicate of #3142
